user-lock-manager.smartapp.groovy

```diff
@@ -1066,3 +1119,69 @@ private sendMessage(msg) {
     sendNotificationEvent(msg)
   }
 }
+
+def checkRecentUser() {
+  debugLog("checkRecentUse: Checking state.cookieCurrentUser: ${state}")
+  if ( state.cookieCurrentUser ) {
+    debugLog "Returned from user ${state.cookieCurrentUser} settings page"
+    def id = state.cookieCurrentUser
+    if ( ! id.isNumber() ) {
+    	debugLog("${id} is NOT a number")
+        id = state.cookieCurrentUser.toInteger()
+    } 
+    if ( id > 0 ) {
+	debugLog("Processing slot #$id")
+	processUserCode(id)
+    }
+  }
+}
+
+def processUserCode(id) {
```

Comment:
Yeah, I actually knew that. I figured I would just add that as a 
function to facilitate adding the handling of schedules later. As it 
turned out, the polling was actually disabling the user pretty soon 
after the lock was updated.

The primary reason I had the lock getting updated right away was to 
speed up the time between changing/enabling/disabling the codes. We are 
using your app to manage a property where people rent the downstairs for 
events. The folks that use the app are of an older generation and when 
they make a change, they expected it to be ready by the time they got to 
the door to test. I got tired of getting calls saying it didn't work 
when all they had to do was try it again a minute later.

I also believe I have a way to do individual scheduling of lock codes 
without adding any additional scheduled jobs. One of the primary 
motivations for the state cookie. I have spec'd it out but, as I like to 
say, 'life happened' and I have not had even a minute to work on it in 
weeks.



------ Original Message ------
From: "Erik Thayer" <notifications@github.com>
To: "ethayer/user-lock-manager" <user-lock-manager@noreply.github.com>
Cc: "A.J." <ajfasano@gmail.com>
Sent: 5/29/2015 4:58:56 PM
Subject: Re: [user-lock-manager] Device type update, page refresh, code 
update, and slot choice for multiple instances (#11)

>In user-lock-manager.smartapp.groovy:
>
> > + debugLog("checkRecentUse: Checking state.cookieCurrentUser: 
>${state}") > + if ( state.cookieCurrentUser ) { > + debugLog "Returned 
>from user ${state.cookieCurrentUser} settings page" > + def id = 
>state.cookieCurrentUser > + if ( ! id.isNumber() ) { > + 
>	debugLog("${id} is NOT a number") > + id = 
>state.cookieCurrentUser.toInteger() > + } > + if ( id > 0 ) { > 
>+	debugLog("Processing slot #$id") > +	processUserCode(id) > + } > + } 
> > +} > + > +def processUserCode(id) {
>This function doesn't take into account schedules, and will activate a 
>code that otherwise should not be active. I would allow the regular 
>initialization and polling interval to take care of this for us instead 
>of initializing users individually.
>
>—
>Reply to this email directly or view it on GitHub.
>